components/SegmentedControl/src/Segmented/Segmented.cs

```diff
@@ -40,18 +26,18 @@ protected override bool IsItemItsOwnContainerOverride(object item)
     protected override void OnApplyTemplate()
     {
         base.OnApplyTemplate();
+        SelectedIndex = _internalSelectedIndex;
```

Comment:
Thinking we should probably only do this once? If the template is re-applied, then the `SelectedIndex` may have changed elsewhere after the initial load? Maybe it's something we just test/keep an eye on later?

Reply:
Good catch - added a hasloaded check, that should fix this?